develop/smart-contracts/overview.md

```diff
@@ -1,4 +1,145 @@
+
+On one end, developers can leverage WASM-based smart contracts chains that support [ink!](TODO:update-path){target=\_blank}, a Rust-based smart contract language that acts as an embedded domain-specific language (eSDL) for Rust. This allows developers to benefit from Rust’s syntax, type safety, memory safety, and extensive tooling.
+
+Conversely, developers can deploy EVM smart contracts on Ethereum-compatible chains that Polkadot secures. Due to EVM compatibility, developers can use any smart contract language that compiles to EVM bytecode, like [Solidity](https://soliditylang.org/){target=\_blank} and [Vyper](https://vyperlang.org/){target=\_blank}. Additionally, popular wallets like [MetaMask](https://metamask.io/){target=\_blank} can be connected and developers can leverage tools they are familiar with, like [Remix](https://remix.ethereum.org/){target=\_blank}, [Hardhat](https://hardhat.org/){target=\_blank}, [Foundry](https://getfoundry.sh/){target=\_blank}, among others.
+
+This guide highlights some differences between building smart contract applications or parachains within Polkadot's scalable and interoperable framework. It also explores the different development paths for smart contract developers on Polkadot.
+
+!!!info "Parachain Developer?" 
+    If you are a Parachain developer looking to add smart contract functionality to your chain, please check the following sections for [WASM-based](TODO:update-path){target=\_blank} or [EVM-based contracts](TODO:update-path){target=\_blank}.
+
+## Smart Contracts Versus Parachains
+
+In a smart contract, you create a program that executes specific logic isolated to the chain on which it is being executed. All the logic executed is bound to the same state transition rules given by the underlying virtual machine (VM). Consequently, smart contracts are more streamlined to develop, and programs can easily interact with each other through similar interfaces.
+
+``` mermaid
+flowchart LR
+    A[(Chain State)] --> B[[Virtual Machine]]
+    C[Transaction] --> B
+    D["Program Logic and Storage<br/>(Smart Contract)"] --> B
+    B --> E[(New State)]
+    B --> F[Execution Logs]
+```
+
+In addition, because smart contracts are programs that execute on top of existing chains, teams don't have to think about the underlying consensus they are built on. 
+
+However, these strengths come with certain limitations. Contracts are immutable by default. Hence, developers have developed different [proxy strategies](https://blog.openzeppelin.com/proxy-patterns){target=\_blank} to be able to upgrade them over time. Generally speaking, such patterns rely on a proxy contract, where the program storage lives, forwarding a call to an implementation contract, and where the execution logic resides. Therefore, smart contract upgrades require changing the implementation contract while retaining the same storage structure, necessitating careful planning. 
```

Comment:
This only applies for EVM-based smart contracts. The pallet contracts has a native implementation to upgrade the logic